### DISCUSSION TITLE
Hidden Hardware

### DISCUSSION THREAD

**Discussion by winterscar** [toverainc/willow#236] [General]
Hi! I'm busy renovating my new home, and one of the things I'd love to do is integrate discrete microphones throughout the house running willow. Something similar to these ['architectural microphones' from josh.ai](https://joshdotai.medium.com/josh-ai-unveils-josh-nano-the-worlds-first-architectural-microphone-bd7e95197655).

The ESP-S3-Box seems like a good starting point for this, but without the screen. Does anyone know if the screen and case can be removed without affecting performance?

**Comment by nikito** [comment]
I think part of the problem is that the microphones are usually
arranged/optimized in the case to achieve the best performance for Far
field, AEC, Beam Forming, etc. By removing them you may affect how they are
"tuned" which could negatively impact performance. That said it may be
possible to create a case that has similar structure but in a different
form factor (something kind of like the ones linked from josh.ai, where it
is a flat surface that can be mounted inside a wall, and the two ports for
the microphones are spaced the right distance apart). The ESP32-S3 box is
just an esp32 inside with some additional hardware so I imagine you can
tinker with it. You could also potentially create a sort of "wrapper"
container that goes around the ESP box so that it is flush to the wall but
is mounted with the majority of its body inside so it isn't sticking out.
Just some ideas! :)
Thanks and Best Regards,
Nick Bento

**Reply by winterscar** [reply]
Thanks Nick! I guess I'll have to pick one up and try tinkering with it :)
